subsys/fs/zffs/src/zffs_file.c

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright (c) 2018 Findlay Feng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zffs/block.h"
+#include "zffs/file.h"
+#include "zffs/misc.h"
+#include "zffs/object.h"
+
+#define file_is_write(__file) ((__file)->write.block.id != ZFFS_NULL)
+#define file_write_free_size(__file) \
+	(ZFFS_CONFIG_BLOCK_SIZE -    \
+	 ((__file)->offset & (ZFFS_CONFIG_BLOCK_SIZE - 1)))
+
+struct file_dlist_node_disk {
+	zffs_disk(u32_t, block);
+};
+
+int zffs_file_make(struct zffs_data *zffs,
+		   struct zffs_area_pointer *pointer, struct zffs_dir *dir,
+		   const struct zffs_node_data *node_data,
+		   struct zffs_file *file)
+{
+	int rc;
+
+	rc = zffs_dir_append(zffs, pointer, dir, node_data);
+	if (rc) {
+		return rc;
+	}
+
+	file->id = node_data->id;
+	file->size = 0;
+	file->offset = 0;
+	file->next_id = node_data->file.next_id;
+	zffs_dlist_init(&file->list);
+	file->block.id = ZFFS_NULL;
+	file->node.id = ZFFS_NULL;
+	file->node.prev = ZFFS_NULL;
+	file->node.next = ZFFS_NULL;
+	file->write.block.id = ZFFS_NULL;
+
+	sys_slist_append(&zffs->opened, (void *)file);
+	return 0;
+}
+
+int zffs_file_open(struct zffs_data *zffs,
+		   struct zffs_area_pointer *pointer,
+		   const struct zffs_node_data *node_data,
+		   struct zffs_file *file)
+{
+	int rc;
+
+	if (node_data->type != ZFFS_TYPE_FILE) {
+		return -EISDIR;
+	}
+
+	file->id = node_data->id;
+	file->size = node_data->file.size;
+	file->next_id = node_data->file.next_id;
+	file->offset = 0;
+	file->list.wait_update = false;
+	file->list.head = node_data->file.head;
+	file->list.tail = node_data->file.tail;
+	file->write.block.id = ZFFS_NULL;
+
+	if (zffs_dlist_is_empty(&file->list)) {
+		file->block.id = ZFFS_NULL;
+		file->node.id = ZFFS_NULL;
+		file->node.prev = ZFFS_NULL;
+		file->node.next = ZFFS_NULL;
+	} else {
+		struct file_dlist_node_disk disk;
+
+		rc = zffs_dlist_head(zffs, pointer, &file->list, &file->node);
+		if (rc < 0) {
+			return rc;
+		}
+
+		if (rc < sizeof(disk)) {
+			return -EIO;
+		}
+
+		rc = zffs_area_read(zffs, pointer, &disk, sizeof(disk));
+		if (rc) {
+			return rc;
+		}
+
+		rc = zffs_block_open(zffs, pointer, sys_get_le32(disk.block),
+				     &file->block);
+		if (rc < 0) {
+			return rc;
+		}
+
+		if (rc != ZFFS_CONFIG_BLOCK_SIZE) {
+			return -EIO;
+		}
+
+		file->pointer = *pointer;
+	}
+
+	sys_slist_append(&zffs->opened, (void *)file);
+
+	return 0;
+}
+
+static int file_sync(struct zffs_data *zffs,
+		     struct zffs_area_pointer *pointer,
+		     struct zffs_file *file)
+{
+	int rc;
+
+	if (file_is_write(file)) {
```

Comment:
change this to:
if  (!file_is_write(file)) { 
return 0;
}

Code below won't be that much indented then. easier to read

Reply:
yep